src/api/sfc-script-setup.md

```diff
@@ -0,0 +1,268 @@
+---
+sidebarDepth: 1
+---
+
+# SFC `<script setup>`
+
+`<script setup>` is a compile-time syntactic sugar for using [Composition API](https://v3.vuejs.org/api/composition-api.html) inside Single File Components (SFCs). It is the recommended syntax if you are using both SFCs and Composition API. It provides a number of advantages over the normal `<script>` syntax:
+
+- More succinct code with less boilerplate
+- Ability to declare props and emitted events using pure TypeScript
+- Better runtime performance (template is compiled into render function in the same scope without an intermediate proxy)
+- Better IDE type-inference performance (less work for the language server to extract types from code)
+
+## Basic Syntax
+
+To opt-in to the syntax, add the `setup` attribute to the `<script>` block:
+
+```html
+<script setup>
+  console.log('hello script setup')
```

Comment:
I noticed all the JS code inside `<script setup>` of SFC `<script setup>` docs are indented 2 spaces while there is no indent in the rest of docs.

Is this something intentional with `<script setup>`?

Reply:
Good catch, prettier actually treats embedded `vue` and `html` blocks differently and these code blocks should be using `vue` instead.